src/OidcClient.php

```diff
@@ -99,11 +103,29 @@ public function refreshTokens(string $refreshToken): OidcTokens
 
     // Request and verify the tokens
     return $this->verifyTokens(
-      $this->requestTokens('refresh_token', null, null, $refreshToken),
+      new OidcTokens(
+        $this->requestTokens('refresh_token', null, null, $refreshToken)
+      ),
       verifyNonce: false
     );
   }
 
+  public function exchangeTokens(string $accessToken, ?string $targetScope = null, ?string $targetAudience = null): OidcTokens
+  {
+    // Clear session after check
+    $this->sessionStorage->clearState();
+
+    // Request exchange tokens. Exchange tokens have no ID token, so no verification here
```

Comment:
Should we try to validate the new access token?

Reply:
That's not true, it only doesn't do anything when the access token isn't a JWT (as that format isn't required by the spec). But if it is a JWT, it is validated.

Reply:
You are right of course. I added the access token validation.